src/NeoSharp.BinarySerialization/Extensions/BinaryWriterExtensions.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System.Text;
+
+namespace System.IO
+{
+    public static class BinaryWriterExtensions
+    {
+        public static void WriteBytesWithGrouping(this BinaryWriter writer, byte[] value)
+        {
+            const int groupSize = 16;
+            var index = 0;
+            var remain = value.Length;
+            while (remain >= groupSize)
+            {
+                writer.Write(value, index, groupSize);
+                writer.Write((byte)0);
+                index += groupSize;
+                remain -= groupSize;
+            }
+            if (remain > 0)
+                writer.Write(value, index, remain);
+            var padding = groupSize - remain;
+            for (var i = 0; i < padding; i++)
+                writer.Write((byte)0);
+            writer.Write((byte)padding);
+        }
+
+        public static void WriteFixedString(this BinaryWriter writer, string value, int length)
+        {
+            if (value == null)
+                throw new ArgumentNullException(nameof(value));
+            if (value.Length > length)
```

Comment:
why can't we always count on value.length?

Reply:
This method is moved, i think that length is maxLength. I think that is not necessary, do you want to remove it?